h2/src/main/org/h2/command/dml/Insert.java

```diff
@@ -388,8 +388,8 @@ private boolean handleOnDuplicate(DbException de, Value[] currentRow) {
         Expression[] row = (currentRow == null) ? valuesExpressionList.get((int) getCurrentRowNumber() - 1)
                 : new Expression[columns.length];
         for (int i = 0; i < columns.length; i++) {
-            String key = table.getSchema().getName() + "." +
-                    table.getName() + "." + columns[i].getName();
+            StringBuilder builder = table.getSQL(new StringBuilder()).append('.');
+            String key = columns[i].getSQL(builder).toString();
```

Comment:
a little bit lower down, we pass key into session.setVariable

are you sure all of the call sites are using quoted names for this functionality?

Reply:
This code is not used in Regular mode, it's a MySQL compatibility stuff. Performance is not very critical aspect here.

This change was performed for safety, names of columns may have own dots, MySQL allows it too and such names are actually used in some projects. Second part of this change in is `Function` class. There are no other users of these variables. See `TestDuplicateKeyUpdate` for code samples.

Reply:
I think that conflict in the name of session variable here due to weird identifiers is very unlikely, but from time to time new issue reports appear due to missing `quoteIdentifier()` invocations.

It will be simpler to perform code review in this area when there are no callers of `getName()` methods that concatenate them with other strings.